### DISCUSSION TITLE
Differences between 'Booking Made or Approuved Emails' and 'Booking Reminder Email'

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#485] [Q&A] [answered]
Hi,
In the 'Booking Made or Approuved Emails' in 'Booking Made Email Body' format, I have a line like this
```
[eme_if tag='#ESC_IS_USER_MEMBER_OF{5}' value='1'] Paiement par Avoir : #_FIELD{21}, il vous reste [eme_mymemberinfo membership_id=5 template_id=86] [/eme_if] 
```
#_FIELD{21} is Dropdown generic custom field with value Oui ou Non.

This format works well in the 'Booking Made or Approuved Emails' but not in 'Accepted Booking Reminder Email'. 

Why this difference?

**Comment by liedekef** [comment]
The placeholder  #_IS_USER_MEMBER_OF{xx} is only to compare if the **current logged-in user** is a member of xx. This is not meant to be used in emails of course.

**Reply by f14m07** [reply, marked as answer]
Ok. If I have understood correctly, the first email works because the recipient is logged in. But the reminder email leaving at another time without anyone being logged in is empty.

**Comment by liedekef** [comment]
Btw: the next version will have #_IS_PERSON_MEMBER_OF{xx} , which will check if the current EME person being handled is a member of said groups. That will in fact be more what you need

**Reply by f14m07** [reply]
Thank you for this new opportunity. If I understand correctly, I will then have to create a group identical to the membership.

**Reply by liedekef** [reply]
No, it will work identical to #_IS_USER_MEMBER_OF , but for an EME person

**Reply by f14m07** [reply]
I did the test in the reminder email but it didn't work. 
```
[eme_if tag='#ESC_IS_PERSON_MEMBER_OF{9}' value='1'] Paiement par Avoir : #_FIELD{21}, il vous reste [eme_mymemberinfo membership_id=5 template_id=86] [/eme_if]
```
It should have said 'Paiement par Avoir...' but nothing appears, I'm looking for

**Reply by f14m07** [reply]
I have this in php log errors:
Warning: Undefined variable $wp_id in /Users/fredericmoriniere/Local Sites/voguer/app/public/wp-content/plugins/events-made-easy/eme-people.php on line 478

**Reply by liedekef** [reply]
correct, but it wasn't tested yet :-) Try the latest eme-people.php now

**Reply by f14m07** [reply]
no more php log error. But I did'nt get the result I hope. I will test again tomorrow. May be I need also to update the other php file. Thanks a lot for your quick responses

**Reply by liedekef** [reply]
I've reworked the code a bit more. At first the check only worked for people linked to a wp user, now it should work for all EME people with an EME membership. The files changed are eme-people.php and eme-members.php

**Reply by f14m07** [reply]
Thanks, I wasn't in. I'll download the dev version and install it locally to check. All the people are WP users and members. I'll be back later to give you the results

**Reply by f14m07** [reply]
This line in the format of 'Accepted Booking Reminder Email Body' does not appear to be read:
[eme_if tag='#ESC_IS_PERSON_MEMBER_OF{9}' value='1'] Paiement par Avoir : #_FIELD{21}, [eme_mymemberinfo membership_id=5 template_id=86] [/eme_if]
This would be normal if the test is negative. However, the person does belong to group '9' and membership '5'.
No php error.

**Reply by liedekef** [reply]
Like I said: IS_PERSON_MEMBER_OF works identical to #_IS_USER_MEMBER_OF, so it checks for membership not groups.

**Reply by liedekef** [reply]
The dev-version now also has #_IS_PERSON_IN_GROUP{xx} (with xx a group id)

**Comment by f14m07** [comment]
Now I understand. Substituting the membership id instead of the group id works. I have a last issue with [eme_mymemberinfo membership_id=5 template_id=86], the template_id=86 contains a member custom field. So maybe it does not work in an event email ?

**Reply by liedekef** [reply]
eme_mymemberinfo again only works for the current logged in user.
Again a dev update: the next version will have a new shortcode eme_person_memberinfo (see the readme in the dev version)

**Reply by f14m07** [reply]
thank you for all these developments

**Reply by f14m07** [reply]
I tested the latest short code and it works very well. Thank you very much for this development, which has enabled me to solve my problem.

**Reply by liedekef** [reply]
Did you download the complete dev-version and test that? That would help me too :-)

**Reply by f14m07** [reply]
No I have only upload the eme-people.php eme-members.php and eme-shotcodes.php. But I will upload the dev version and test

**Reply by f14m07** [reply]
At that moment no problem with the EME dev version. Creation of event with EMEFS, RSVP, mails, all works as expected

**Reply by liedekef** [reply]
Thanks, I'm testing it too on a public site and it works fine too :-)

**Comment by f14m07** [comment]
I have this warning in php log:
Warning: Undefined variable $output in /Users/fredericmoriniere/Local Sites/voguer/app/public/wp-content/plugins/events-made-easy/eme-members.php on line 4424

**Reply by liedekef** [reply]
That is because of the wrong usage of eme_mymemberinfo ... but the dev version will solve that warning
